Data/Aeson/Parser/Time.hs

```diff
@@ -46,13 +46,18 @@ run p t = case A.parseOnly (p <* endOfInput) t of
             Left err -> fail $ "could not parse date: " ++ err
             Right r  -> return r
 
--- | Parse a date of the form @YYYY-MM-DD@.
+-- | Parse a date of the form @[+,-]YYYY-MM-DD@.
 day :: Parser Day
 day = do
+  absOrNeg <- choice
+    [ char '-' *> pure negate
```

Comment:
(in fact `negate <$ char '-' <|> negate <$ char '+" <|> pure id` may inline better, as `choice` is recursive, but again, probably there are no difference.